web/renderer/components/pageComponents/ConnectionsPage/NewConnection/About.tsx

```diff
@@ -1,58 +1,191 @@
-          });
-        }}
-        options={[
-          { label: "MySQL/Dolt", value: DatabaseType.Mysql },
-          {
-            label: "Postgres/Doltgres",
-            value: DatabaseType.Postgres,
-          },
-        ]}
-        hideSelectedOptions
-        light
-      />
-      <Button type="submit" disabled={!state.name} className={css.button}>
-        Next
-      </Button>
+      {!startDoltServer && (
+        <FormSelect
+          outerClassName={css.typeSelect}
+          className={css.typeSelectInner}
+          labelClassName={css.label}
+          label="Type"
+          val={state.type}
+          onChangeValue={t => {
+            if (!t) return;
+            setErr(undefined);
+            setState({
+              type: t,
+              port: t === DatabaseType.Mysql ? "3306" : "5432",
+              username: t === DatabaseType.Mysql ? "root" : "postgres",
+            });
+          }}
+          options={[
+            { label: "MySQL/Dolt", value: DatabaseType.Mysql },
+            {
+              label: "Postgres/Doltgres",
+              value: DatabaseType.Postgres,
+            },
+          ]}
+          hideSelectedOptions
+          light
+        />
+      )}
+      {startDoltServer && (
+        <FormInput
+          label="Port"
+          value={state.port}
+          onChangeString={p => {
+            setState({ port: p });
+            setErr(undefined);
+          }}
+          placeholder="Enter port number"
+          light
+          labelClassName={css.label}
+        />
+      )}
+      <ButtonsWithError error={err || connectErr}>
+        {startDoltServer ? (
+          <>
+            <Button
+              type="submit"
+              disabled={
+                getStartLocalDoltServerDisabled(state, storedConnections)
+                  .disabled
+              }
+              className={css.button}
+              onClick={handleSubmit}
+              data-tooltip-content={
+                getStartLocalDoltServerDisabled(state, storedConnections)
+                  .message
+              }
+              data-tooltip-id="add-local-dolt-server"
+            >
+              Start and Connect to Dolt Server
+            </Button>
+            <Tooltip
+              id="add-local-dolt-server"
+              className={css.tooltip}
+              place="bottom"
+            />
+          </>
+        ) : (
+          <Button type="submit" disabled={!state.name} className={css.button}>
+            Next
+          </Button>
+        )}
+      </ButtonsWithError>
     </form>
   );
 }
+
+type DisabledReturnType = {
+  disabled: boolean;
+  message?: string;
+};
+
+function getStartLocalDoltServerDisabled(
+  state: ConfigState,
+  connections?: DatabaseConnectionFragment[],
+): DisabledReturnType {
+  const disabled =
+    !state.name ||
+    !state.port ||
+    !!connections?.some(connection => connection.isLocalDolt);
+
```

Comment:
Can return early here if not `disabled` so you don't have to go through all these checks